service/matching/matching_engine.go

```diff
@@ -226,11 +235,67 @@ func (e *matchingEngineImpl) Stop() {
 		return
 	}
 
+	e.serviceResolver.RemoveListener(e.listenerKey())
+	close(e.membershipChangedCh)
+
 	for _, l := range e.getTaskQueues(math.MaxInt32) {
 		l.Stop()
 	}
 }
 
+func (e *matchingEngineImpl) listenerKey() string {
+	return fmt.Sprintf("matchingEngine[%p]", e)
+}
+
+func (e *matchingEngineImpl) watchMembership() {
+	self := e.hostInfoProvider.HostInfo().Identity()
+
+	for range e.membershipChangedCh {
+		delay := e.config.MembershipUnloadDelay()
+		if delay == 0 {
```

Comment:
Why do we do nothing when this value is zero? I'm assuming it's to disable the new behavior, but we should document that alongside the dynamicconfig so it's explicit

Reply:
yup, added comment